mesonbuild/compilers/compilers.py

```diff
@@ -631,6 +631,9 @@ def get_exelist(self, ccache: bool = True) -> T.List[str]:
     def get_linker_exelist(self) -> T.List[str]:
         return self.linker.get_exelist() if self.linker else self.get_exelist()
 
+    def get_exe_file(self) -> str:
+        return os.path.realpath(self.exelist_no_ccache[0])
```

Comment:
This returns a bad value when only the executable name is passed (so shouldn't `shutil.which()` be used here?)